Index: net/http/http_cache_transaction.cc
===================================================================
--- http_cache_transaction.cc   (revision 241641)
+++ http_cache_transaction.cc   (working copy)
@@ -2240,7 +2240,8 @@
   // reverse-map the cert status to a net error and replay the net error.
   if ((cache_->mode() != RECORD &&
        response_.headers->HasHeaderValue("cache-control", "no-store")) ||
-      net::IsCertStatusError(response_.ssl_info.cert_status)) {
+       (!cache_->GetSession()->params().ignore_certificate_errors &&
+       net::IsCertStatusError(response_.ssl_info.cert_status))) {
     DoneWritingToEntry(false);
     ReportCacheActionFinish();
     if (net_log_.IsLoggingAllEvents())